### PR TITLE
Add tag cli args for pkg repo release

### LINKED ISSUE #1639: `kctrl package repository release` Should Allow Control of Image Tags
**Describe the problem/challenge you have**
Currently, there is no way to control the image tags for created and published package repository imgpkg bundles.  This prevents the creation of a package repository with a version that includes any build metadata, e.g. `0.0.1+git.c9cc39a`

**Describe the solution you'd like**
The `kctrl package repository release` command should gain a `--tag` flag that is the equivalent of the `kctrl package release` command's `--tag` flag.

**Anything else you would like to add:**
[Slack](https://kubernetes.slack.com/archives/CH8KCCKA5/p1731518888544399)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by devacts**
#### Which issue(s) this PR fixes:

Fixes #1639

#### Does this PR introduce a user-facing change?

```release-note
Allow Control of Image Tags with introduction of `--tag` flag in `kctrl package repo release`
```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: cli/test/e2e/package_repo_release_test.go [hunk 1]
```diff
@@ -58,6 +58,23 @@ func TestPackageRepositoryReleaseInteractively(t *testing.T) {
...
+			promptOutput.WaitFor("Enter the registry url")
+			promptOutput.Write(env.Image)
+		}()
+
+		kctrl.RunWithOpts([]string{"pkg", "repo", "release", "--tty=true", "--chdir", workingDir, "--version", "1.0.0", "--tag", "build-tag-0001"},
+			RunOpts{NoNamespace: true, StdinReader: promptOutput.StringReader(),
+				StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
+
+		keysToBeIgnored := []string{"creationTimestamp:", "image"}
+		verifyPackageRepoBuild(t, keysToBeIgnored)
```
Comment (praveenrewar) [inline cli/test/e2e/package_repo_release_test.go:84]: Let's also validate that the tag got used.

Comment (devacts) [reply cli/test/e2e/package_repo_release_test.go:84]: Validation added in the commit: 5f31c9ded4501dd86c0dc05a4c82a0f32c37f7ab

File: cli/test/e2e/package_repo_release_test.go [hunk 2]
```diff
@@ -49,13 +50,47 @@ func TestPackageRepositoryReleaseInteractively(t *testing.T) {
...
 				StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
 
 		keysToBeIgnored := []string{"creationTimestamp:", "image"}
 		verifyPackageRepoBuild(t, keysToBeIgnored)
 		verifyPackageRepository(t, keysToBeIgnored)
+
+		args := []string{"tag", "list", "-i", os.Getenv("KCTRL_E2E_IMAGE")}
+		cmd := exec.Command("imgpkg", args...)
+		output, err := cmd.Output()
+		// t.Logf("the tags are: %s", output)
```
Comment (praveenrewar) [inline cli/test/e2e/package_repo_release_test.go]: nit
```suggestion
```

File: cli/test/e2e/package_repo_release_test.go [hunk 3]
```diff
@@ -49,13 +50,47 @@ func TestPackageRepositoryReleaseInteractively(t *testing.T) {
...
+				StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
+
+		keysToBeIgnored := []string{"creationTimestamp:", "image"}
+		verifyPackageRepoBuild(t, keysToBeIgnored)
+		verifyPackageRepository(t, keysToBeIgnored)
+
+		args := []string{"tag", "list", "-i", os.Getenv("KCTRL_E2E_IMAGE")}
+		cmd := exec.Command("imgpkg", args...)
+		output, err := cmd.Output()
+		// t.Logf("the tags are: %s", output)
```
Comment (praveenrewar) [inline cli/test/e2e/package_repo_release_test.go]: nit
```suggestion
```

**Review by praveenrewar — APPROVED** [review]
LGTM!

File: cli/test/e2e/package_repo_release_test.go (hunk 2 above)
Comment (devacts) [reply cli/test/e2e/package_repo_release_test.go]: removed in commit: c65ba79a7092f834b8bb5109b86b7942ef04e58b

File: cli/test/e2e/package_repo_release_test.go (hunk 3 above)
Comment (devacts) [reply cli/test/e2e/package_repo_release_test.go]: removed in commit: c65ba79a7092f834b8bb5109b86b7942ef04e58b

**Pull request merged**